src/components/Nav/Nav.js

```diff
@@ -1,7 +1,182 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { Link } from 'react-router-dom';
+import EventBar from './EventBar';
+import SearchedResult from './SearchedResult';
 
 const Nav = () => {
-  return <div>Nav</div>;
+  const [searchedTerm, setSearchedTerm] = useState('');
+  const [isEventBarVisible, setIsEventBarVisible] = useState(true);
+  const [scrollY, setScrollY] = useState(0);
+
+  const saveTerm = e => {
+    setSearchedTerm(e.target.value);
+  };
+
+  const clearTerm = () => {
+    setSearchedTerm('');
+  };
+
+  const handleScroll = () => {
+    setScrollY(window.scrollY);
+  };
+
+  useEffect(() => {
+    scrollY > 50 ? setIsEventBarVisible(false) : setIsEventBarVisible(true);
+  }, [scrollY]);
+
+  window.addEventListener('scroll', handleScroll);
+
+  return (
+    <Container>
+      {isEventBarVisible && (
+        <EventBar setIsEventBarVisible={setIsEventBarVisible} />
+      )}
+      <ContentsWrap>
+        <CenterWrap>
+          <Link to="/">
+            <Logo src="./images/logo.png" />
+          </Link>
+          <MenuItems>
+            <MenuItem>커뮤니티</MenuItem>
+            <MenuItem>스토어</MenuItem>
+          </MenuItems>
+        </CenterWrap>
+        <SearchSet>
+          <SearchBarSet>
+            <Icon src="./images/search.png" />
+            <SearchBar
+              onChange={saveTerm}
+              value={searchedTerm}
+              placeholder="아이디, 상품 검색"
+            />
+            {searchedTerm.length !== 0 ? (
+              <XIcon onClick={clearTerm} src="./images/close.png" />
+            ) : null}
```

Comment:
논리곱 단축평가를 활용하시면 될 것 같습니다.
```suggestion
            {searchedTerm.length !== 0 && (
              <XIcon onClick={clearTerm} src="./images/close.png" />
            )}
```